### DISCUSSION TITLE
Livewire Select2 Dynamic not updating public view

### DISCUSSION THREAD

**Discussion by Vladjen** [livewire/livewire#3543] [1. Help] [answered]
I am using a select2 component with `wire:ignore` and i want to update the select2 value dynamically after clicking a button. I set up this functionality with events and events work fine, so does the variable gets initialized as well. I am failing to update this public view of this select2.

my blade
```php
<select class="select2-example form-control" id="subjects" wire:model.lazy="subjects"  name="subjects"> 
 </select> 

@push('scripts')
<script>
$('#subjects').select2({
        maximumSelectionLength: 1,
        minimumInputLength:2,        
        tags: false,
        placeholder: 'Enter Subject(s)',
       .... // this all works great
});   

$('#subjects').on('change', function (e) {
        let data = $(this).val();
        @this.set('subjects', data);
});

// my event listener and it is working as well
Livewire.on('editSubject', subject => {
         console.log(subject);

         @this.set('subjects', subject);
         $('#subjects').val(subject);
         $('#subjects').trigger('change');   //the public view doesn't get updated
}) 
</script>
@endpush
```
I so far tried with browser dispatch event as well. Nothing works. What would be the workaround for this? Any help is greatly appreciated.

**Comment by joshhanley** [comment, marked as answer]
@Vladjen have a look at this issue #1890 as it should give you some ideas of how to fix it. Always worth searching though old issues and discussions history to see if there is any information 🙂

**Reply by Vladjen** [reply]
Thank you for the clarification. I managed to fix the issue. @joshhanley

**Reply by joshhanley** [reply]
@Vladjen no worries! 🙂
